File: debezium/connector/KafkaConnectorClient.java
package com.tiket.tix.hotel.common.debezium.connector;

import java.io.IOException;
import okhttp3.MediaType;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.RequestBody;
import okhttp3.Response;

public class KafkaConnectorClient {
  private final OkHttpClient client = new OkHttpClient();

  public void registerConnector(String kafkaConnectUrl, String connectorName, String jsonConfig) throws IOException {
    RequestBody body = RequestBody.create(MediaType.get("application/json"), jsonConfig);
    Request request = new Request.Builder()
        .url(kafkaConnectUrl + "/connectors")
        .post(body)
        .build();

    try (Response response = client.newCall(request).execute()) {
      if (response.isSuccessful()) {
        System.out.println("Connector " + connectorName + " registered");
      } else {
        System.err.println("Failed to register connector " + connectorName + ": " + response.body().string());
      }
    }
  }
}

File: debezium/internal/connector/kafka/kafka_connector_client.go
package kafka

import (
	"bytes"
	"fmt"
	"io/ioutil"
	"net/http"
)

type KafkaConnectorClient struct {
	client *http.Client
}

func NewKafkaConnectorClient() *KafkaConnectorClient {
	return &KafkaConnectorClient{
		client: &http.Client{},
	}
}

func (k *KafkaConnectorClient) RegisterConnector(kafkaConnectURL, connectorName, jsonConfig string) error {
	url := fmt.Sprintf("%s/connectors", kafkaConnectURL)
	body := bytes.NewBuffer([]byte(jsonConfig))

	req, err := http.NewRequest("POST", url, body)
	if err != nil {
		return fmt.Errorf("failed to create request: %w", err)
	}
	req.Header.Set("Content-Type", "application/json")

	resp, err := k.client.Do(req)
	if err != nil {
		return fmt.Errorf("failed to register connector %s: %w", connectorName, err)
	}
	defer resp.Body.Close()

	respBody, _ := ioutil.ReadAll(resp.Body)
	if resp.StatusCode >= 200 && resp.StatusCode < 300 {
		fmt.Printf("Connector %s registered\n", connectorName)
	} else {
		return fmt.Errorf("failed to register connector %s: %s", connectorName, string(respBody))
	}

	return nil
}
